ppci/format/elf/file.py

```diff
@@ -8,16 +8,104 @@
 import logging
 
 from ...arch.arch_info import Endianness
-from .headers import ElfMachine, HeaderTypes
+from .headers import (
+    ElfMachine,
+    HeaderTypes,
+    SectionHeaderType,
+    SymbolTableBinding,
+    SymbolTableType,
+)
 
 
 logger = logging.getLogger("elf")
 
 
+class ElfRelocation:
+    def __init__(self, header, bits=64):
+        self.header = header
+        self.bits = bits
+        self.parse_info()
+
+    def __getitem__(self, key):
+        if key == "type":
+            return self.type
+        elif key == "symbol_id":
+            return self.symbol_id
+        elif key == "section":
+            return self.section
+        elif key == "offset":
+            return hex(self.header.r_offset)
+        elif key == "addend":
+            if self.header.r_addend:
+                return hex(self.header.r_addend)
+            else:
+                return hex(0)
+        else:
+            raise IndexError
+
+    def parse_info(self):
+        if self.bits == 64:
+            self.symbol_id = self.header.r_info >> 32
+            self.type = self.header.r_info & 0xFFFFFFFF
+        else:
+            self.symbol_id = self.header.r_info >> 8
+            self.type = self.header.r_info & 0xFF
+
+    def connect_section(self, symbole_table, sections):
+        symbole = symbole_table[self.symbol_id].header
+        self.section = sections[symbole.st_shndx].name
+
+
+class ElfSymbol:
+    def __init__(self, header, i):
+        self.header = header
+        self.i = i
+        self.parse_info()
+
+    def __getitem__(self, key):
+        if key == "binding":
+            return SymbolTableBinding(self.binding).name.lower()
+        elif key == "id":
+            return self.i
+        elif key == "name":
+            return self.name
+        elif key == "section":
+            return self.section.name
+        elif key == "size":
+            return self.header.st_size
+        elif key == "typ":
+            return SymbolTableType(self.type).name.lower()
+        elif key == "value":
+            return str(hex(self.header.st_value))
+        else:
+            raise IndexError
+
+    def parse_info(self):
+        self.binding = self.header.st_info >> 4
+        self.type = self.header.st_info & 0xF
+
+
 class ElfSection:
     def __init__(self, header):
         self.header = header
 
+    def __getitem__(self, key):
```

Comment:
Rather than making the conversion here to objectfile, I would make this conversion in a separate function.